src/main/java/mops/presentation/DirectoryController.java

```diff
@@ -0,0 +1,128 @@
+package mops.presentation;
+
+import lombok.AllArgsConstructor;
+import mops.businesslogic.Account;
+import mops.businesslogic.DirectoryService;
+import mops.businesslogic.FileQuery;
+import mops.businesslogic.FileService;
+import mops.businesslogic.utils.AccountUtil;
+import mops.persistence.Directory;
+import mops.persistence.FileInfo;
+import org.keycloak.adapters.springsecurity.token.KeycloakAuthenticationToken;
+import org.springframework.data.repository.query.Param;
+import org.springframework.stereotype.Controller;
+import org.springframework.ui.Model;
+import org.springframework.web.bind.annotation.*;
+
+import java.util.List;
+
+@Controller
+@RequestMapping("/material1/dir")
+@AllArgsConstructor
+public class DirectoryController {
+    /**
+     * Manages all directory queries.
+     */
+    private final DirectoryService directoryService;
+
+    /**
+     * Manges all file queries.
+     */
+    private final FileService fileService;
+
+    /**
+     * @param token keycloak auth token
+     * @param model spring view model
+     * @param dirId id of the folder
+     * @return route to folder
+     */
+    @GetMapping(path = "/{dirId}")
```

Comment:
`path = ` should not be necessary here. did you try without?

Reply:
yes it works without it, copy & paste it from documentation